### PR TITLE
Allow string functions to be called with arguments of datatypes that were supported earlier

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
While fixing issue in string functions for input containing multibyte characters, we blocked string functions called with some datatypes to be more compliance with TSQL, which were supported earlier by using underlying PostgreSQL string functions. This PR include changes to allow all string functions to be called on all the datatypes that were support earlier.

Following are the string functions which faced this issue and will be fixed with this PR:- 
```
CONCAT, CONCAT_WS
STRING_AGG
TRIM, LTRIM, RTRIM, LEFT, RIGHT
REPLACE
REPLICATE
REVERSE
TRANSLATE
STUFF
SUBSTRING
LOWER and UPPER
```

### Issues Resolved

BABEL-5337

### Test Scenarios Covered ###
* **Use case based -** YES 

* **Boundary conditions -** YES

* **Arbitrary inputs -** NA

* **Negative test cases -** NA

* **Minor version upgrade tests -** NA

* **Major version upgrade tests -** NA

* **Performance tests -** NA

* **Tooling impact -** NA

* **Client tests -** YES

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -1016,7 +1016,7 @@ class tsqlCommonMutator : public TSqlParserBaseListener
 						}
 					}
 				}
-				if (id->keyword()->TRIM() || id->keyword()->REPLACE() || id->keyword()->TRANSLATE() || id->keyword()->SUBSTRING() || id->keyword()->STRING_AGG()  || id->keyword()->CONCAT() || id->keyword()->CONCAT_WS())
+				if (id->keyword()->TRIM() || id->keyword()->REPLACE() || id->keyword()->TRANSLATE() || id->keyword()->SUBSTRING() || id->keyword()->STRING_AGG())
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: can you please explain this change?

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 2]
```diff
@@ -1314,6 +1090,16 @@ tsql_func_select_candidate_for_special_func(List *names, List *fargs, int nargs,
 	if (!is_func_validated)
 		return NULL;
 
+	new_input_typeids = (Oid *) palloc(nargs * sizeof(Oid));
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c]: better to use palloc0 otherwise there could be valid garnage Oid which may bypass OidIsValid check somewhere

File: test/JDBC/expected/db_collation/ISC-Check-Constraints-vu-verify.out [hunk 3]
```diff
@@ -74,7 +74,7 @@ isc_check_constraints_db1#!#dbo#!#test_tsql_const_c_varbinary_check#!#((c_varbin
 isc_check_constraints_db1#!#dbo#!#test_udd_c_udd_check#!#((c_udd < 133.230182309832423))
 isc_check_constraints_db1#!#dbo#!#test_udd_c_udd1_check#!#((c_udd1 < '2020-10-20 09:00:00'))
 isc_check_constraints_db1#!#dbo#!#test_udd_c_udd2_check#!#((c_udd2 < '09:00:00'))
-isc_check_constraints_db1#!#dbo#!#test_upper_a_check#!#((sys.upper(a) IN ('A', 'B')))
+isc_check_constraints_db1#!#dbo#!#test_upper_a_check#!#((upper(a) IN ('A', 'B')))
```
Comment (Deepesh125) [inline test/JDBC/expected/db_collation/ISC-Check-Constraints-vu-verify.out:77]: why this diff?

File: test/JDBC/upgrade/14_8/schedule [hunk 4]
```diff
@@ -448,9 +448,9 @@ babel_4328_datetimeoffset-before-16_3
 reverse-before-15_8-or-16_4
 stuff-before-15_8-or-16_4
 replicate-before-15_8-or-16_4
-string_agg-before-15_9-or-16_5
+string_agg-before-15_5-or-14_10
 concat-before-15_9-or-16_5
-concat_ws-before-15_9-or-16_5
+concat_ws-before-15_5-or-14_100
```
Comment (Anikait143) [inline test/JDBC/upgrade/14_8/schedule]: please replace `14_100` -> `14_10`

File: test/python/expected/upgrade_validation/expected_dependency.out [hunk 5]
```diff
@@ -78,7 +78,12 @@ Function sys."nchar"(integer)
 Function sys."nchar"(sys."nchar",integer,boolean)
 Function sys."nchar"(sys.varbinary)
 Function sys."session_user"()
+Function sys."substring"(bit,integer,integer)
+Function sys."substring"(bytea,integer,integer)
 Function sys."system_user"()
+Function sys."trim"(bytea,bytea)
+Function sys."trim"(text)
+Function sys."trim"(text,text)
```
Comment (Anikait143) [inline test/python/expected/upgrade_validation/expected_dependency.out]: Can we have dependency tests for these?

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 6]
```diff
@@ -1018,96 +1003,13 @@ get_immediate_base_type_of_UDT(PG_FUNCTION_ARGS)
...
 bool
 validate_special_function(char *func_nsname, char *func_name, List* fargs, int nargs, Oid *input_typeids, bool num_args_match)
 {
 	tsql_special_function_t    *special_func;
-	bool                        type_match;
-	Oid                         input_type_id, valid_type_id, base_type_id;
-	Oid                         sys_varcharoid;
 
 	/* Sanity checks */
 	if (func_name == NULL || (nargs != 0 && input_typeids == NULL) || fargs == NIL)
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c]: We don't require `input_typeids` and `fargs` as the function parameters anymore. So, we should remove them from function definition

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 2 above)
Comment (Anikait143) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Can we have a comment specifying the purpose of this block (checking for UDTs)?

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 7]
```diff
@@ -1322,52 +1108,43 @@ tsql_func_select_candidate_for_special_func(List *names, List *fargs, int nargs,
...
-			|| (*common_utility_plugin_ptr->is_tsql_nchar_datatype)(input_typeids[2]))
+		if ((*common_utility_plugin_ptr->is_tsql_nvarchar_datatype)(new_input_typeids[0])
+			|| (*common_utility_plugin_ptr->is_tsql_nchar_datatype)(new_input_typeids[0])
+			|| (*common_utility_plugin_ptr->is_tsql_ntext_datatype)(new_input_typeids[0])
+			|| (*common_utility_plugin_ptr->is_tsql_nvarchar_datatype)(new_input_typeids[1])
+			|| (*common_utility_plugin_ptr->is_tsql_nchar_datatype)(new_input_typeids[1])
+			|| (*common_utility_plugin_ptr->is_tsql_ntext_datatype)(new_input_typeids[1])
+			|| (*common_utility_plugin_ptr->is_tsql_nvarchar_datatype)(new_input_typeids[2])
+			|| (*common_utility_plugin_ptr->is_tsql_nchar_datatype)(new_input_typeids[2])
+			|| (*common_utility_plugin_ptr->is_tsql_ntext_datatype)(new_input_typeids[1]))
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c]: should be `new_input_typeids[2]` instead

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 1 above)
Comment (Anikait143) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp]: I understand that this change is required because we can't make certain datatypes to hit the `variadic` definition, so we also consider PG definitions here, but i think we can do that with other functions mentioned in this list too, it will help maintain uniformity and the behaviour would be same as before for datatypes other than the ones mentioned in TSQL doc.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql [hunk 8]
```diff
@@ -10589,6 +10589,924 @@ $$
...
+DECLARE
+    exception_message text;
+BEGIN
+	CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'replicate_any_deprecated_in_5_0_0'); 
+EXCEPTION WHEN OTHERS THEN
+    GET STACKED DIAGNOSTICS
+    exception_message = MESSAGE_TEXT;
+    RAISE WARNING '%', exception_message;
+END;
+$$;
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql]: Function `sys.babelfish_drop_deprecated_object` itself raises the warning in case of exception, we can directly invoke it instead of using DO blocks. We can modify `sys.babelfish_drop_deprecated_object` definition to cover all error cases. (Applies to all other places as well) https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/a106cb18c5891a38b9b7f7248b16b313474c90f7/contrib/babelfishpg_common/sql/upgrades/babelfish_common_helper--4.3.0--4.4.0.sql#L33

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 9]
```diff
@@ -3228,104 +3198,43 @@ CREATE OR REPLACE FUNCTION sys.substring(string TEXT, i INTEGER, j INTEGER)
...
+CREATE OR REPLACE FUNCTION sys.substring(string PG_CATALOG.BIT, i INTEGER, j INTEGER)
+RETURNS PG_CATALOG.BIT
+AS 'bitsubstr' LANGUAGE INTERNAL IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.substring(string sys.VARCHAR, i INTEGER, j INTEGER)
-RETURNS sys.VARCHAR
-AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE OR REPLACE FUNCTION sys.substring(string BYTEA, i INTEGER, j INTEGER)
+RETURNS BYTEA
+AS 'bytea_substr' LANGUAGE INTERNAL IMMUTABLE STRICT PARALLEL SAFE;
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: Why do we need dedicated definitions for BIT/BYTEA, shouldn't we let PG handle them?

**Review by Anikait143 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 1 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp]: Removed all the other functions, except `TRIM` since if we did not append sys to `TRIM()` function then it will always hit PostgreSQL `btrim` function from the grammer.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 2 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Added

File: test/JDBC/upgrade/14_8/schedule (hunk 4 above)
Comment (rohit01010) [reply test/JDBC/upgrade/14_8/schedule]: Updated

File: test/python/expected/upgrade_validation/expected_dependency.out (hunk 5 above)
Comment (rohit01010) [reply test/python/expected/upgrade_validation/expected_dependency.out]: Added

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 6 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Removed.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 7 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Yeah, updated.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql (hunk 8 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql]: Modified the `sys.babelfish_drop_deprecated_object()` function to cover this undefined function scenario.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 9 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Removed

File: test/JDBC/upgrade/16_6/schedule [hunk 10]
```diff
@@ -533,7 +533,7 @@ BABEL-4707
 BABEL_4817
 upper_lower
 BABEL-4641
-trim
+trim-before-15_11-or-16_7
```
Comment (Deepesh125) [inline test/JDBC/upgrade/16_6/schedule:536]: why isnt there schedule file for 16_5?

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql [hunk 11]
```diff
@@ -10589,6 +10592,712 @@ $$
...
+    exception_message text;
+BEGIN
+    ALTER FUNCTION sys.replicate(ANYELEMENT, INTEGER) RENAME TO replicate_any_deprecated_in_5_0_0;
+EXCEPTION WHEN OTHERS THEN
+    GET STACKED DIAGNOSTICS
+    exception_message = MESSAGE_TEXT;
+    RAISE WARNING '%', exception_message;
+END;
+$$;
+DO $$
```
Comment (sumitj824) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql:10656]: NIT: Should we combine all of such rename statement in single DO block since either all of them will be present or absent?

File: test/JDBC/expected/db_collation/ISC-Check-Constraints-vu-verify.out (hunk 3 above)
Comment (rohit01010) [reply test/JDBC/expected/db_collation/ISC-Check-Constraints-vu-verify.out:77]: the column `a` of table `test_upper` is of type `sys.BPCHAR`, previously we had a definition of `upper()` function as `sys.UPPER(sys.BPCHAR)` so we were getting `sys.` prepended. But now as we had removed that definition `upper()` is not getting prepended with `sys` here.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 12]
```diff
@@ -3228,104 +3198,35 @@ CREATE OR REPLACE FUNCTION sys.substring(string TEXT, i INTEGER, j INTEGER)
...
 CREATE OR REPLACE FUNCTION sys.substring(string sys.NVARCHAR, i INTEGER, j INTEGER)
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.substring(string sys.NCHAR, i INTEGER, j INTEGER)
-RETURNS sys.NVARCHAR
-AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
-
-CREATE OR REPLACE FUNCTION sys.substring(string sys.VARBINARY, i INTEGER, j INTEGER)
+CREATE OR REPLACE FUNCTION sys.substring(string sys.bbf_varbinary, i INTEGER, j INTEGER)
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql:3209]: What will be the behaviour in case PG binary types such as `BYTEA`?

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 13]
```diff
@@ -1018,99 +1003,16 @@ get_immediate_base_type_of_UDT(PG_FUNCTION_ARGS)
...
-validate_special_function(char *func_nsname, char *func_name, List* fargs, int nargs, Oid *input_typeids, bool num_args_match)
+validate_special_function(char *func_nsname, char *func_name, int nargs, bool num_args_match)
 {
 	tsql_special_function_t    *special_func;
-	bool                        type_match;
-	Oid                         input_type_id, valid_type_id, base_type_id;
-	Oid                         sys_varcharoid;
 
 	/* Sanity checks */
-	if (func_name == NULL || (nargs != 0 && input_typeids == NULL) || fargs == NIL)
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c:1113]: Why we had to remove the check on `nargs`?

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 14]
```diff
@@ -1486,7 +1313,6 @@ tsql_func_select_candidate_for_special_func(List *names, List *fargs, int nargs,
 
 static FuncCandidateList
 tsql_func_select_candidate(List *names,
-						   List *fargs,
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c]: I don't think it's a good idea to remove the additional parameter we are getting from PG, as it would require an additional Engine commit + if we might require this later during our development process then we will have to redo this. It would do no harm being here.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c [hunk 15]
```diff
@@ -1458,20 +1279,26 @@ tsql_func_select_candidate_for_special_func(List *names, List *fargs, int nargs,
 			continue;
 
 		rettype = get_func_rettype(current_candidate->oid);
+		
+		/* Ignore following definitions as these are used when no other potential definition can be used. */
+		if ((current_candidate->args[0] == TEXTOID && rettype == get_sys_varcharoid())
+			|| (current_candidate->args[0] == BITOID && rettype == BITOID)
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Redundant: We don't have any definitions for `BIT` datatype

File: test/JDBC/expected/concat_ws-before-15_5-or-14_10-vu-prepare.out [hunk 16]
```diff
@@ -437,13 +417,15 @@ INSERT INTO babel_3409_concat_ws_t2 VALUES
     N'المتقين'
 )
 GO
+~~ROW COUNT: 3~~
+
 
 CREATE VIEW babel_3409_concat_ws_dep_view1 AS
-    SELECT CONCAT_WS('-', col_varchar, col_char, col_varbinary, col_text) AS concatenated_column FROM babel_3409_concat_ws_t1
+    SELECT CONCAT_WS('-', col_varchar, col_int, col_varbinary, col_text) AS concatenated_column FROM babel_3409_concat_ws_t1
```
Comment (Anikait143) [inline test/JDBC/expected/concat_ws-before-15_5-or-14_10-vu-prepare.out:424]: Why this change?

File: test/JDBC/expected/concat-vu-verify.out
Comment (Anikait143) [inline test/JDBC/expected/concat-vu-verify.out:1]: Some of the function calls which were originally returning `TEXT` due to PG definitions are now returning `VARCHAR`/`NVARCHAR` as the return type because of TSQL definitions. Have we verified that it is only the return type change and not the output difference?

File: test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/right-vu-verify.out [hunk 17]
```diff
@@ -4,14 +4,14 @@ SELECT @custname,
     TRIM(@custname) as [TRIM]
     , LTRIM(@custname) as [LTRIM]
     , RTRIM(@custname) as [RTRIM]
-    , LEFT(@custname, 4) as [LEFT4]
+    , RIGHT(@custname, 4) as [RIGHT4]
```
Comment (Anikait143) [inline test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/right-vu-verify.out]: Why this change?

File: test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/rtrim-vu-verify.out [hunk 18]
```diff
@@ -2,7 +2,7 @@
 DECLARE @custname NVARCHAR(50) = N'比尔·拉莫斯'
 SELECT @custname, 
     TRIM(@custname) as [TRIM]
-    , LTRIM(@custname) as [LTRIM]
+    , RTRIM(@custname) as [RTRIM]
     , RTRIM(@custname) as [RTRIM]
```
Comment (Anikait143) [inline test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/rtrim-vu-verify.out]: same as above.

File: test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/translate-vu-verify.out [hunk 19]
```diff
@@ -621,25 +621,30 @@ varchar
 DECLARE @inputString CHAR(50) = '  abc🙂defghi🙂🙂    ', @characters VARCHAR(20) = 'ab?', @translations CHAR(20) = 'x🙂y'
 SELECT '|' + TRANSLATE(@inputString, @characters, @translations) + '|'
 GO
+~~START~~
+varchar
 ~~ERROR (Code: 33557097)~~
 
-~~ERROR (Message: The second and third arguments of the TRANSLATE built-in function must have same argument data type.)~~
+~~ERROR (Message: The second and third arguments of the TRANSLATE built-in function must contain an equal number of characters.)~~
```
Comment (Anikait143) [inline test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/translate-vu-verify.out:628]: Can you check if we threw an error earlier as well with PG definitions for such cases?

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 12 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql:3209]: Refer test case added where the argument type is PG binary type `BYTEA` [link](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/97ff19130dd8aa96dca14675f5a1fe9dab7cec97/test/JDBC/expected/substring-vu-verify.out#L1066)

File: test/JDBC/upgrade/16_6/schedule (hunk 10 above)
Comment (rohit01010) [reply test/JDBC/upgrade/16_6/schedule:536]: After `16.4`, the next version we have is `16.6`. Hence we do not have `16.5` schedule file.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 13 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c:1113]: previously we had a check `(nargs != 0 && input_typeids == NULL)` to make sure that `input_typeids` is not null when `nargs` is 0, now since we are not using `input_typeids` in `validate_special_function`, we do not need this check.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 14 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: Added back.

File: contrib/babelfishpg_tsql/src/pltsql_coerce.c (hunk 15 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/pltsql_coerce.c]: removed

File: test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/right-vu-verify.out (hunk 17 above)
Comment (rohit01010) [reply test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/right-vu-verify.out]: it was done by mistake, changed back.

File: test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/translate-vu-verify.out (hunk 19 above)
Comment (rohit01010) [reply test/JDBC/expected/non_default_server_collation/chinese_prc_ci_as/translate-vu-verify.out:628]: Earlier when we were using PG definitions for translate, we were allowing 2nd and 3rd argument with different lengths, as PG supports it [link](https://www.postgresql.org/docs/9.1/functions-string.html#:~:text=Any%20character%20in%20string%20that%20matches%20a%20character%20in%20the%20from%20set%20is%20replaced%20by%20the%20corresponding%20character%20in%20the%20to%20set.%20If%20from%20is%20longer%20than%20to%2C%20occurrences%20of%20the%20extra%20characters%20in%20from%20are%20removed.). But in TSQL above error is thrown when 2nd and 3rd argument are with different length, so we will be following the same behaviour here.

**Review by Anikait143 — APPROVED** [review]
LGTM! Just please revert https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3192/commits/2848a21ed9add4df2f7017706364d4320800126e

File: test/JDBC/expected/concat-vu-verify.out
Comment (rohit01010) [reply test/JDBC/expected/concat-vu-verify.out:1]: There will be some output difference as the return type is changed, although the new output is either same as previous or is similar as TSQL behaviour.

**Review by Anikait143 — APPROVED** [review]

**Review by sumitj824 — APPROVED** [review]

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql (hunk 11 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.4.0--5.0.0.sql:10656]: No, we should keep these statements independent, this way will avoid the possiblity of non-existent of some functions which can be dropped by user.

**Review by rishabhtanwar29 — DISMISSED** [review]

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]
